File: create_tables/ppdm_quality_control.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table ppdm_quality_control';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table ppdm_quality_control (
        SYSTEM_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        TABLE_NAME VARCHAR(30) NOT NULL,
        QC_SEQ_NO INT,
        ACTIVE_IND VARCHAR(1),
        CHECKED_BY_BA_ID VARCHAR(40),
        COLUMN_NAME VARCHAR(30),
        CURRENT_DATE_VALUE TIMESTAMP(0),
        CURRENT_NUMERIC_VALUE DOUBLE PRECISION,
        CURRENT_NUMERIC_VALUE_OUOM VARCHAR(40),
        CURRENT_NUMERIC_VALUE_UOM VARCHAR(40),
        CURRENT_ROW_GUID VARCHAR(38),
        CURRENT_TEXT_VALUE VARCHAR(2000),
        DATA_TYPE VARCHAR(40),
        DONE_BY_BA_ID VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        NULL_DESCRIPTION VARCHAR(1024),
        PPDM_GUID VARCHAR(38),
        QC_DATETIME TIMESTAMP(0),
        QC_DESC VARCHAR(1024),
        QC_STATUS VARCHAR(40),
        QC_TYPE VARCHAR(40),
        QUALITY_TYPE VARCHAR(40),
        REMARK VARCHAR(2000),
        RETENTION_PERIOD VARCHAR(40),
        SOURCE VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/PpdmQualityControl.go
package dto

type Ppdm_quality_control struct {
	System_id                  string   `json:"system_id" default:"source"`
	Table_name                 string   `json:"table_name" default:"source"`
	Qc_seq_no                  int      `json:"qc_seq_no" default:"1"`
	Active_ind                 *string  `json:"active_ind" default:""`
	Checked_by_ba_id           *string  `json:"checked_by_ba_id" default:""`
	Column_name                *string  `json:"column_name" default:""`
	Current_date_value         *string  `json:"current_date_value" default:""`
	Current_numeric_value      *float64 `json:"current_numeric_value" default:""`
	Current_numeric_value_ouom *string  `json:"current_numeric_value_ouom" default:""`
	Current_numeric_value_uom  *string  `json:"current_numeric_value_uom" default:""`
	Current_row_guid           *string  `json:"current_row_guid" default:""`
	Current_text_value         *string  `json:"current_text_value" default:""`
	Data_type                  *string  `json:"data_type" default:""`
	Done_by_ba_id              *string  `json:"done_by_ba_id" default:""`
	Effective_date             *string  `json:"effective_date" default:""`
	Expiry_date                *string  `json:"expiry_date" default:""`
	Null_description           *string  `json:"null_description" default:""`
	Ppdm_guid                  *string  `json:"ppdm_guid" default:""`
	Qc_datetime                *string  `json:"qc_datetime" default:""`
	Qc_desc                    *string  `json:"qc_desc" default:""`
	Qc_status                  *string  `json:"qc_status" default:""`
	Qc_type                    *string  `json:"qc_type" default:""`
	Quality_type               *string  `json:"quality_type" default:""`
	Remark                     *string  `json:"remark" default:""`
	Retention_period           *string  `json:"retention_period" default:""`
	Source                     *string  `json:"source" default:""`
	Row_changed_by             *string  `json:"row_changed_by" default:""`
	Row_changed_date           *string  `json:"row_changed_date" default:""`
	Row_created_by             *string  `json:"row_created_by" default:""`
	Row_created_date           *string  `json:"row_created_date" default:""`
	Row_effective_date         *string  `json:"row_effective_date" default:""`
	Row_expiry_date            *string  `json:"row_expiry_date" default:""`
	Row_quality                *string  `json:"row_quality" default:""`
}
